File: java/problems/p0069_sqrt_x/Solution.java
package problems.p0069_sqrt_x;

class Solution {
    public int mySqrt(int x) {
        if (x == 0 || x == 1) {
            return x;
        }

        int a = 0;
        int b = x / 2 + 1;
        int mid;

        while (b - a > 1) {
            mid = (a + b) / 2;

            if ((long)mid*mid <= x) {
                a = mid;
            } else {
                b = mid;
            }
        }

        return a;
    }
}

File: typescript/problems/p0069_sqrt_x/solution.ts
function mySqrt(x: number): number {
  if (x === 0 || x === 1) {
    return x;
  }

  let a: number = 0;
  let b: number = Math.floor(x / 2) + 1;
  let mid: number;

  while (b - a > 1) {
    mid = Math.floor((a + b) / 2);

    if (mid * mid <= x) {
      a = mid;
    } else {
      b = mid;
    }
  }

  return a;
}

export default mySqrt;
